ui/thread-view.html

```diff
@@ -57,9 +57,51 @@
 </div>
 
 <div id="body_template" class="body_part">
-  <iframe class="body_iframe" sandbox srcdoc=""></iframe>
+  <!--
+    sandbox="allow-same-origin" is required, otherwise the scrollHeight of the
+    embedded document cannot be accessed.
+    Do not add 'allow-scripts' to the sandbox attribute. This would allow the
+    document to execute scripts itself.
+  -->
+  <iframe class="body_iframe" sandbox="allow-same-origin" onload="handleLoadIframe(this);" srcdoc=""></iframe>
 </div>
 
+<script>
```

Comment:
There's also a `WebKitSettings:enable-javascript-markup` setting you can use to remove any doubt; that allows the application to run JavaScript but blocks the document from doing so. That would block the use of `<script>` here, but you could probably run it manually using webkit_web_view_evaluate_javascript().

Reply:
This seems good for (faith in) security. But I did not see how I can reference `this` (the iframe) in cpp in a way such that I can pass it to the javascript function `handleLoadIframe`.

At least, I expect it to be a bit complicated.

Are your doubts strong, currently?

Reply:
> But I did not see how I can reference this (the iframe) in cpp in a way such that I can pass it to the javascript function handleLoadIframe.

The best solution is the reverse: don't try to access it from C++. Move work that touches the DOM into JavaScript instead. The C++ DOM APIs are all deprecated and already gone in the webkitgtk-6.0 API version, so they'll be a problem for you in the future.

> Are your doubts strong, currently?

No. I _think_ the sandbox attribute is good enough because you're careful to encode everything so it can't break out. I think you're good so long as you use `Glib::Markup::escape_text` on all the untrusted input, and your comment says you do that.

[This guideline](https://cheatsheetseries.owasp.org/cheatsheets/DOM_based_XSS_Prevention_Cheat_Sheet.html#guideline-4-avoid-sending-untrusted-data-into-html-rendering-methods) suggests you don't use inner HTML at all, but if all untrusted input is properly encoded then I'm not sure why not.

Reply:
Thanks! Yes, in the long run, this schould be done, maybe one can just pass the emails as json to the javascript somehow, (anyway, it seems you can only pass data to js by injecting js strings, so it would be good to have *dead* json) and then use js to build the whole threadview. But this one is even more work.